### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by AnatoleAM** [SevenTV/Extension#344] [General]
#### High-Performance Chat

<span style="font-size:larger;">
We've made chat run faster, by rebuilding it from the ground up. Let your processor and memory relax, with 7TV your chat experience will be much more efficient.
</span>

[cgl_display_chat.webm](https://user-images.githubusercontent.com/9599344/225709498-3a6e6182-b3dc-4f89-bc56-79749ee2780b.webm)

#### New Emote Menu

<span style="font-size:larger;">The emote menu was completely redesigned, it's now divided into providers, emote sets, and even shows emojis! It also got smarter, only loading the images currently in view for much better performance.</span>

![image](https://user-images.githubusercontent.com/9599344/225709695-ced58a07-f9db-44e3-95e0-0ae369333523.png)

#### New Settings Menu

<span style="font-size:larger;">7TV has just gotten much more customizable, check out the new menu and tune all the various settings to your liking!</span>

![image](https://user-images.githubusercontent.com/9599344/225709741-d169c58f-42b8-41d4-84bd-51dc0a5501e6.png)

#### Personal Emotes

<span style="font-size:larger;">7TV Subscribers can now use Personal Emotes in chat! These are emotes that can be used globally, but only by you. Get your own by <a href="https://7tv.app/store" target="_blank">subscribing</a>.</span>

![image](https://user-images.githubusercontent.com/9599344/225709820-056fc713-07e6-4dc3-b9d0-d53088faa77f.png)

#### Dynamic Cosmetics

<span style="font-size:larger;">Badges, Paints & Animated Avatars will now update instantly. That's right, no more wait times!</span>

![image](https://user-images.githubusercontent.com/9599344/225709963-8e449dcd-134d-46ae-88e8-ada3e63c1fd4.png)

#### Custom Chat Highlights

<span style="font-size:larger;">Set up custom highlights for specific phrases in chat messages, with a color, label, title flash and custom sounds. Use Regular Expressions for even greater control.</span>

![image](https://user-images.githubusercontent.com/9599344/225710055-c27a373a-0b9f-48e1-b98d-40a736b2a8a4.png)

#### Tab-Completion Carousel

<span style="font-size:larger;">View matches as you go when using TAB to find emotes!</span>

![image](https://user-images.githubusercontent.com/9599344/225710095-e041edf0-f69e-455d-ba8d-5572977ad6bf.png)

#### AVIF Support

<span style="font-size:larger;">
We are the first emote provider to support WEBP, and as we've inspired others to do the same, we're taking the initiative of supporting AVIF (AV1 Image File Format)!
AVIF is the image variant of the AV1 Video Codec, capable of shrinking image sizes from 20% to 90% over a GIF. 
</span>

AVIF will now be automatically served to users of Google Chrome 100+ (or chromium-based browsers).

![image](https://user-images.githubusercontent.com/9599344/225710176-730066c9-aa1d-4d8c-9839-5de5f6addde9.png)

#### Mod Logs

<span style="font-size:larger;">View recent mod actions with the Mod Logs menu.</span>

#### Much more to come

<span style="font-size:larger;">This is just the beginning! We've rewritten the entire extension with a modular, scalable codebase just waiting to be expanded upon. Expect many new additions to come!</span>

And of course, this is an open-source project. Anyone is welcome to submit a pull request via <a href="https://github.com/seventv/extension" target="_blank">GitHub</a>.

<hr /><em>This discussion was created from the release <a href='https://github.com/SevenTV/Extension/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by dubbyOW** [comment]
I would recommend to add a setting that allows colon emote completion to sort emotes with a rule "Prioritize emotes that start with user input", this is one of the settings in FFZ extension, currently it doesnt work, because 7tv overrides it, therefore I had to remove 7tv, since disabling colon completion on 7tv also disabled it for FFZ, example of why its problematic:
I type ":ome" and the 1st emote instead of "OMEGALUL", can be something like "BBoomer", because its sortet in alphabetical order and "BBoomer" contains "ome" in the name, this is an example of trying to use colon completion in xQc chat, if you want to see it yourself

additionally also add a rule "Prioritize favorite emotes at the top", also from FFZ, it works in a way that the more you use an emote, lets say "OMEGALUL" in this case, even if theres more emotes starting with "ome", then "OMEGALUL" would be on top and easy access with a single Enter tap for input

I have a habit of typing ":ome" and pressing Enter for a quick "OMEGALUL" emote, this build of 7TV extension, took this feature away
